warehouse/index.js

```diff
@@ -10,6 +9,8 @@ const whPageConfigJson = require("./config/WHPageConfig.json");
 const whScreenConfigJson = require("./config/WHScreenConfig.json");
 const whGroupConfigJson = require("./config/WHGroupConfig.json");
 const whAliasConfigJson = require("./config/WHAliasConfig.json");
+// redshift destination string limit, if the string length crosses 512 we will change data type to text which is varchar(max) in redshift
+const RSStringLimit = 512;
```

Comment:
This file should be generic, if we add destination specific code in this then it becomes hard to read/maintain going forward.

Reply:
One approach could be to change the signature of processSingleMessage.

```
*** warehouse/index.js
--  function processWarehouseMessage(provider, message, schemaVersion, RSAlterStringToText = "false") 

++ function processWarehouseMessage(provider, message, options = {}) 

-- const getDataType = (val)

++ const getDataType = (val, options) => {
  if options.getDataType && typeof options.getDataType === "function" {
     return options.getDataType(val)
  }
}
```

```
*** rs/transform.js
++ const RSStringLimit = 512;

++ func getDataType(val) {
++ // Do redshift specific handling here. Pass this function to warehouse index.js
++ }

-- processWarehouseMessage("rs", message, schemaVersion, RSAlterStringToText);

++ processWarehouseMessage("rs", message, {schemaVersion, getDataType});
```